src/redash_stmo/integrations/iodide/extension.py

```diff
@@ -0,0 +1,62 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, you can obtain one at http://mozilla.org/MPL/2.0/.
+import requests
+import json
+import logging
+import os
+
+from redash.models import Query
+from redash.handlers.base import BaseResource, get_object_or_404
+from redash.permissions import require_permission
+
+from redash_stmo import settings
+from redash_stmo.resources import add_resource
+
+
+logger = logging.getLogger(__name__)
+
+
+class IodidePostResource(BaseResource):
+    @require_permission("view_query")
+    def get(self, query_id):
+        query = get_object_or_404(
+            Query.get_by_id,
+            query_id,
+        )
+        url = "https://stage.iodide.nonprod.dataops.mozgcp.net/api/v1/notebooks/"
+        template = open(
+            "/extension/src/redash_stmo/integrations/iodide/iodide-template.iomd",
```

Comment:
A few things here:

1. Could you please rename the file to `iodide-notebook.iomd.j2`?

2. This shouldn't hard-code the file system path to the template.

   Use this: `os.path.join(os.path.dirname(__file__), 'iodide-template.iomd.j2')`.

3. Redash uses Jinja2 for templating, let's use this here, too, to profit from autoescaping and other things that Jinja2 provides.

```python
from flask import render_template_string

with open(os.path.join(os.path.dirname(__file__), 'iodide-notebook.iomd.j2'), "r") as template:
    source = template.read()
    context = {
        "query_id": query_id,
        "title": query.name,
        "api_key": settings.REDASH_OWN_API_KEY_FOR_IODIDE,
    }
    completed_template = render_template_string(source, context)
```

4. Of course this means you'll need to use Jinja2/Django style curly braces for the placeholders in `iodide-notebook.iomd.j2`, instead of `[<NAME>]`.

Reply:
Ah, good suggestion. Thanks!